pkg/eval/pwd_test.go

```diff
@@ -0,0 +1,40 @@
+package eval
+
+import (
+	"errors"
+	"path/filepath"
+	"testing"
+)
+
+func TestBuiltinPwd(t *testing.T) {
+	tmpHome, cleanup := InTempHome()
+	defer cleanup()
+
+	mustMkdirAll("dir1")
+	mustMkdirAll("dir2")
+	dir1 := filepath.Join(tmpHome, "dir1")
+	dir2 := filepath.Join(tmpHome, "dir2")
+
+	Test(t,
+		That(`pwd=dir1 put $pwd; put $pwd`).Puts(dir1, tmpHome),
+		That(`cd $E:HOME/dir2; pwd=~ put $pwd; put $pwd`).Puts(tmpHome, dir2),
+		That(`cd $E:HOME/dir2; pwd=~/dir1 put $pwd; put $pwd`).Puts(dir1, dir2),
+		That(`cd $E:HOME/dir1; pwd=../dir2 put $pwd; put $pwd`).Puts(dir2, dir1),
+		That(`pwd=(float64 1) put $pwd`).Throws(ErrPathMustBeString, "pwd=(float64 1)"),
+	)
+}
+
+// Verify the behavior when the CWD cannot be determined.
+func TestBuiltinPwdNoCwd(t *testing.T) {
+	origGetwd := Getwd
+	Getwd = mockGetwd
+	defer func() { Getwd = origGetwd }()
+
+	Test(t,
+		That(`put $pwd`).Puts("/unknown/pwd"),
+	)
+}
+
+func mockGetwd() (string, error) {
+	return "", errors.New("cwd unknown")
+}
```

Comment:
Either rename this to a more descriptive `mockGetwdWithError`, or inline it in the test function above.

Reply:
done